### ISSUE TITLE
Migrate the blog from using plain Markdown files to MDX files instead

### ISSUE THREAD

**Issue by Jarmos-san** [Jarmos-san/jarmos.dev#161]
The blog right now as it is, uses plain Markdown files for storing the prose content which is then rendered into HTML, CSS and JS during the Next.js build process. While this worked well till now, the process is lacking in features and is not capable of making the most out of the React.js ecosystem. This is where MDX will come handy by enabling us to use React components and JSX right inside the Markdown files themselves.

In other words, it will be possible to utilise the [`Link`](https://nextjs.org/docs/app/api-reference/components/link) and [`Image`](https://nextjs.org/docs/app/api-reference/components/image) components for a more optimised experience of the website's contents.

**Comment by Jarmos-san** [comment]
Closing this thread since as a of [the latest release](https://github.com/Jarmos-san/jarmos.dev/releases/latest), the site has been completed rewritten from scratch! And Astro provides excellent builtin Markdown support.

See the release notes for more information in this context.
